### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by mbirman**
Hi guys,

I've done a bunch of small refactorings in the different parts of the app (mainly in controllers).

Please review, comment and merge if it's useful.

Thanks!

### CODE REVIEW

**Comment by herminiotorres** [comment]
@mbirman first could you rebase and resolve this conflicts?

File: app/controllers/contacts_controller.rb [hunk 1]
```diff
@@ -30,4 +29,8 @@ def create
   def set_subject
     @subject = "You have a new message from #{current_user.username} of The Odin Project"
   end
+
+  def set_user
```
Comment (herminiotorres) [inline app/controllers/contacts_controller.rb]: you could create a permit params method? thanks!

File: app/controllers/curriculum_controller.rb [hunk 2]
```diff
@@ -43,14 +28,31 @@ def index
...
-        return api_object.contents.get :path => item.path
+      if item.name =~ /index\.(html|md)/
+        return get_contents(api_object, item.path)
       end
     end
     nil
   end
 
+  def github
+    @github ||= Github::Repos.new(:user => "theodinproject", :repo => "curriculum", :oauth_token => "#{ENV['GITHUB_API_TOKEN']}")
```
Comment (herminiotorres) [inline app/controllers/curriculum_controller.rb]: ruby hash 1.9, please?

File: app/controllers/lesson_completions_controller.rb [hunk 3]
```diff
@@ -1,44 +1,34 @@
...
-    else   
-      @course = @lesson.course
-      @next_lesson = @lesson.next_lesson
-      lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
-      if lc.save
-        render "create", :formats => [:js]
-      else
-        render :nothing => true, :status => 400 # bad request
-      end
+    lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
```
Comment (herminiotorres) [inline app/controllers/lesson_completions_controller.rb]: `lc` for me? is a magic variable, please define a better name for this variable?

**Comment by KevinMulhern** [comment]
Hey @mbirman, there are no unit tests included. I realise this pull request was made over a year ago so you have probably moved on. Sorry we didn't get to this sooner.

**Pull request closed without merging**